boa/src/builtins/array/mod.rs

```diff
@@ -304,6 +305,39 @@ impl Array {
         }
     }
 
+    /// `Array.of(...arguments)`
+    ///
+    /// The Array.of method creates a new Array instance from a variable number of arguments,
+    /// regardless of the number or type of arguments.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-array.of
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/of
+    pub(crate) fn of(this: &Value, args: &[Value], context: &mut Context) -> Result<Value> {
+        let is_constructor = this.is_object() && this.as_object().unwrap().is_constructable();
+
+        // construct using `this` if it is a constructor
+        // otherwise use Array
+        let array = if is_constructor {
+            this.as_object().unwrap().construct(
+                &[Value::from(args.len())],
+                this.clone(),
+                context,
+            )?
+        } else {
+            Array::array_create(args.len() as u32, None, context)?
+        };
+
+        // add properties and set length
+        array.set_field("length", 0, context)?;
```

Comment:
```suggestion
        array.set_field("length", args.len(), context)?;
```

The `length` shouldn't be zero, should be the count of arguments, (`Array.of` spec step 8)

Reply:
The length is set by `Array::add_to_array_object`, setting it to 0 was unnecessary and has been removed